File: src/main/java/com/aquasecurity/plugins/trivy/ui/treenodes/SecretTreeNode.java
package com.aquasecurity.plugins.trivy.ui.treenodes;

import com.aquasecurity.plugins.trivy.icons.TrivyIcons;
import com.aquasecurity.plugins.trivy.model.Location;
import com.aquasecurity.plugins.trivy.model.Secret;

import javax.swing.*;
import javax.swing.tree.DefaultMutableTreeNode;

public class SecretTreeNode extends DefaultMutableTreeNode implements TrivyTreeNode {

    public final Secret secret;
    public final String filepath;

    public SecretTreeNode(Secret secret, String filepath) {
        super(secret);
        this.secret = secret;
        this.filepath = filepath;
    }

    @Override
    public Icon getIcon() {
        return TrivyIcons.Secret;
    }

    @Override
    public String getTitle() {
        return this.secret.ruleID   ;
    }

    @Override
    public String getTooltip() {
        return this.secret.title;
    }

    public Secret getSecret() {
        return secret;
    }

    public Location getLocation() {

        return new Location(filepath, secret.startLine, secret.endLine);
    }
}

File: src/main/kotlin/com/aquasecurity/plugins/trivy/ui/treenodes/SecretTreeNode.kt
package com.aquasecurity.plugins.trivy.ui.treenodes

import com.aquasecurity.plugins.trivy.icons.TrivyIcons
import com.aquasecurity.plugins.trivy.model.Location
import com.aquasecurity.plugins.trivy.model.Secret
import javax.swing.Icon
import javax.swing.tree.DefaultMutableTreeNode

class SecretTreeNode(val secret: Secret, val filepath: String) :
    DefaultMutableTreeNode(secret), TrivyTreeNode {
    val location: Location
        get() = Location(filepath, secret.startLine, secret.endLine)
    override val icon: Icon?
        get() = TrivyIcons.Secret
    override val title: String?
        get() = secret.ruleID
    override val tooltip: String?
        get() = secret.title
}